src/storage/mvcc/reader/util.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use storage::mvcc::{Error, Result};
+use storage::mvcc::{Lock, LockType, Write};
+use storage::{Cursor, Iterator, Key, Snapshot, Statistics, Value, CF_LOCK};
+
+/// Get the lock of a user key in the lock CF.
+///
+/// Internally, a db get will be performed.
+///
+/// You may want to use the wrapper `mvcc::reader::CFReader` instead.
+pub fn load_lock<S>(snapshot: &S, key: &Key, statistics: &mut Statistics) -> Result<Option<Lock>>
+where
+    S: Snapshot,
+{
+    let res = match snapshot.get_cf(CF_LOCK, key)? {
```

Comment:
Is here a `self.statistics.lock.get += 1;` missing?

Reply:
nice catch